Sources/NIOExtras/HTTP1ProxyConnectHandler.swift

```diff
@@ -0,0 +1,226 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2022 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIOCore
+import NIOHTTP1
+
+public final class HTTP1ProxyConnectHandler: ChannelDuplexHandler, RemovableChannelHandler {
+    public typealias OutboundIn = Never
+    public typealias OutboundOut = HTTPClientRequestPart
+    public typealias InboundIn = HTTPClientResponsePart
+
+    enum State {
```

Comment:
`private`?

Reply:
I think these type aliases have to be public because `(NIO)HTTP1ProxyConnectHandler` is public

Reply:
The type aliases certainly have to be public, the `private` comment was for `State`